lib/linalg/Vector.h

```diff
@@ -759,6 +759,27 @@ class Vector
      */
     double localMin(int nmax = 0);
 
+    /**
+     * @brief Distribute this vector among MPI processes,
+     * based on the specified local dimension.
+     * This becomes distributed after this function is executed.
+     *
+     * @pre !distributed()
+     * @pre d_owns_data
+     *
+     * @param[in] local_dim dimension for local MPI rank.
+     */
+    void distribute(const int &local_dim);
+
+    /**
+     * @brief Gather all the distributed elements among MPI processes.
+     * This becomes not distributed after this function is executed.
```

Comment:
Further comment that the data is identical on all processes after this operation?